main.c

```diff
@@ -243,6 +243,12 @@ static const struct wl_callback_listener surface_frame_listener = {
 };
 
 void damage_surface(struct swaylock_surface *surface) {
+	if (surface->state->ext_session_lock_v1 &&
```

Comment:
I think this logic is good to have for layer-shell too, because layer-shell also has the same "don't commit a buffer prior to configure" restrictions.

Reply:
Interesting that it didn’t crash with layer-shell as far as my testing went. Regardless, updated.